src/app/modules/submission/email-submission/email-submission.util.ts

```diff
@@ -520,28 +526,71 @@ export const concatAttachmentsAndResponses = (
   return response
 }
 
-export const maskUidOnLastField = (
-  autoReplyData: EmailAutoReplyField[],
-): EmailAutoReplyField[] => {
+/**
+ * Function to generate email data
+ * for a single field
+ */
+const createFormattedDataForOneField = <T>(
+  response: ProcessedFieldResponse,
+  hashedFields: Set<string>,
+  getFormattedFunction: (
+    response: ResponseFormattedForEmail,
+    hashedFields: Set<string>,
+  ) => T,
```

Comment:
i think you've managed to identify a recurring pattern here. can we extract this into a type?

Something along the lines of

```typescript
type FormattingFunction<T extends ?> = (
    response: ResponseFormattedForEmail,
    hashedFields: Set<string>,
  ) => T
```

Then you can do e.g.

```typescript
const getDataCollationFormattedResponse = (
  response: ResponseFormattedForEmail,
): FormattingFunction<EmailDataCollationToolField | undefined>
```

Reply:
Thanks! I think the generic type `T` should be constrained on `createFormattedDataForOneField` because we also want to constrain the result `T[]`, and not just constraint `T` on getFormattedFunction(). Since it is used only once, I've left it inlined in the function signature - please see if ok

Separately, we also talked about whether the signature of `getFormattedFunction` should have optional `hashedFields?: Set<string>`; this is because some of the callbacks we pass in (`getAutoReplyFormattedResponse` and `getDataCollationFormattedResponse`) do not require the hashedField argument. However, declaring `hashedFields?: Set<string>` means that *all* callbacks must be able to accept only one argument `response`; this is not true for `getFormFormattedResponse`. It turns out that the correct approach in this instance is to write the callback parameters as non-optional - see [here](https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html) under Optional Parameters in Callbacks.

Reply:
`getFormattedFunction` isn't a callback function though. let's have a discussion later?

Reply:
which means this is good!